dali/dfu/dfurun.cpp

```diff
@@ -1358,11 +1373,17 @@ class CDFUengine: public CInterface, implements IDFUengine
                             if (multiclusterinsert||multiclustermerge)
                                 multifdesc.setown(fdesc.getClear());
                             else
+                            {
                                 dstFile.setown(fdir.createNew(fdesc));
+                                IFileDescriptor * srcDesc = (auxfdesc.get() ? auxfdesc.get() : foreignfdesc.get());
+                                if (options->getPreserveCompression() && srcDesc && srcDesc->isCompressed())
+                                    dstFile->queryAttributes().setPropBool("@blockCompressed",true);
```

Comment:
>> old comment :
>>@AttilaVamos - the check should still be before and set the target fdesc (on line 1327) as with other cases.
>> So it should be there that it checks foreignfdesc or auxfdesc if no srcFile is available.

jakesmith 21 minutes ago  Member
@AttilaVamos - your latest commit is still doing it late. See last comment above, re. the change should happen at around line 1327, i.e. you should be changing fdesc (the IFileDescriptor), not later after the desc has been used to create the IDistributedFile

Reply:
this is what I think code should look like (untested):

replace lines 1326-1327
with:
```
    else if (options->getPreserveCompression())
    {
        bool dstCompressed = false;
        if (srcFile)
            dstCompressed = srcFile->isCompressed();
        else
        {
            IFileDescriptor * srcDesc = (auxfdesc.get() ? auxfdesc.get() : foreignfdesc.get());
            dstCompressed = srcDesc && srcDesc->isCompressed();
        }
        if (dstCompressed)
            fdesc->queryProperties().setPropBool("@blockCompressed", dstCompressed);
    }
```